docs/codecs/index.md

```diff
@@ -0,0 +1,113 @@
+# Codecs
+
+TL;DR: Codecs are used to describe how to decode data from the wire and encode it back to wire format.
+
+Tremor connects to external systems using [connectors](../connectors). [Connectors](../connectors) use `codecs` to transform the data Tremor receives from connected system participants into a structured value that forms the payload of each and every Tremor event.
+
+Codecs are the means of turning the (mostly binary) data from the wire (e.g. from a TCP connection) into structured values for Tremor events and back into binary wire format.
+Each [connector](../connectors) can be configured with a `codec`.
+
+## Usage
+
+If you expect [`JSON`](https://www.json.org/) data from a TCP connection, you need to configure the `json` codec.
+
+Example:
+
+```tremor
+define connector tcp_example from tcp_server
+with
+    codec = "json",
+    config = {
+        "url": "localhost:12345"
+    }
+end;
+```
+
+
+This `tcp_example` connector is configured to expect [`JSON`](https://www.json.org/) data from each accepted TCP connection. It expects 1 JSON document after the next without a single byte separating them.
```

Comment:
As we had this issue with #2120 today, i think it would make sense to mention here, that we actually expect 1 valid json document per protocol unit (TCP packet). The sentence above seems to misleed into believing that you could have any number of json documents per TCP packet, but this is not true.

Reply:
It's a fairly bad example, TCP isn't a package based, and there is no guarantee a package send is received as a single package. I'll change it to UDP and clear up the wording